### PR TITLE
Bg-fix-update-profile-issues

### PR DESCRIPTION
**Pull request by aimelive**
### What does this PR do?
- Changed PUT method to PATCH method (User might want to update some profile info, not all).
- User profile is created when user account is created/registered by admin. (basics info initially: email, firstName, lastName, and role. Other fields are empty for now).
- User profile is updated, not created when the end point is called.
- Changed Profile.create() method to Profile.update() in updateProfile() function.
- Added `req.file` condition to check if the user added a file so that the profile would be changed by calling `await fileUpload(req)` method.
- Solved an error `req.file.path` path not found.

### How should this be manually tested?
- After pulling this branch `bg-fix-profile-issues` locally and checkout in it. Do `npm i` to get all dependencies.
- Run `npm run migrate` to migrate all models in the local database
- Run `npm run seed` to populate the tables with seeders' values
- Run `npm run dev` to start the server manually
- Use postman to test the endpoints: 
1. First login as an Admin and create new user POST `api/v1/users/login` and  POST `api/v1/users/register` respectively.
2. To check if user account has been created successfully, do GET `api/v1/users/:id` and  to check his profile info do GET `api/v1/users/:id/profile`.
3. To update/change his/her profile info do PATCH `api/v1/v1/users/:id/update`. N.B: In your req.body you will have informations like this or some:  profilePic, lastName, firstName, nationalId, gender, phone, category, bio, province, district, sector, cell or village.
4. Successfully update will show the message: `User Updated Successfully` otherwise will prompt unsuccessfully message.

### Any background context you want to provide?
When we request for the endpoint, the updateProfile function was creating new Profile, instead of updating the existing one. This issue may cause one user to have more than one row in the profile table. 
### Questions:

### CODE REVIEW

**Review by Kalimbak — APPROVED** [review]

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
Great work here @aimelive. I suggest: 
1. when deleting a user a profile can be deleted too unless there is activating or deactivating an account feature in the app. 
2. You may add profile seeders for the admin that we already have, so far they have null profiles

**Review by UmuhireJessie — APPROVED** [review]

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
Add seeder for our admins' profiles

File: src/controllers/profilecontroller.js [hunk 1]
```diff
@@ -27,100 +21,94 @@ const allUsers = async (req, res) => {
...
-    attributes: {
-      exclude: ['password'],
+      userId: id,
     },
   })
     .then((user) => {
       if (!user) {
         return res.status(400).json({
-          message: req.t('no_user'),
+          message: 'No profile match with this id',
```
Comment (Anguandia) [inline src/controllers/profilecontroller.js]: Why have you removed the translation?

File: src/controllers/profilecontroller.js [hunk 2]
```diff
@@ -27,100 +21,94 @@ const allUsers = async (req, res) => {
...
       message: 'User Updated Successully',
-      data: {
-        userProfile,
-      },
     });
   } catch (error) {
     res.status(404).json({
-      message: 'No user with that ID',
+      message: 'Something went wrong!',
       Error: error.stack,
```
Comment (Anguandia) [inline src/controllers/profilecontroller.js]: do we really need to send the error stack to the user?

**Review by Anguandia — CHANGES_REQUESTED** [review]
Good work, please do the following respond to pending change requests, then we'll be good to go

**Review by Anguandia — APPROVED** [review]

**Review by UmuhireJessie — APPROVED** [review]

**Review by furebo — APPROVED** [review]

**Pull request merged**